### PR TITLE
[jvm] Add ability to load booster direct from byte array

### PR DESCRIPTION
**Pull request by honzasterba**

### CODE REVIEW

**Comment by honzasterba** [comment]
@CodingCat please advise if there is anything else I should do to get this merged, thanks!

**Comment by trivialfis** [comment]
@wbo4958 Could you please help taking a look?

File: jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java [hunk 1]
```diff
@@ -85,11 +85,26 @@ static Booster loadModel(InputStream in) throws XGBoostError, IOException {
       os.write(buf, 0, size);
     }
     in.close();
-    Booster ret = new Booster(new HashMap<String, Object>(), new DMatrix[0]);
```
Comment (wbo4958) [inline jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java:88]: maybe we can delete 

``` java
static Booster loadModel(InputStream in) throws XGBoostError, IOException {
```
and move the logic (getting buffer from InputStream) into XGBoost.java

File: jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java [hunk 2]
```diff
@@ -61,6 +61,20 @@ public static Booster loadModel(InputStream in)
     return Booster.loadModel(in);
```
Comment (wbo4958) [inline jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java:61]: maybe we can add the logic converting InputStream into byte[] here

Comment (CodingCat) [reply jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java:61]: +1 on this

File: jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java (hunk 1 above)
Comment (honzasterba) [reply jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java:88]: done

**Comment by CodingCat** [comment]
merged, thanks!

**Pull request merged**